File: server/features/claude_config.go
package features

import "github.com/tstapler/stapler-squad/server/featureregistry"

// ClaudeConfigGet describes the get-claude-config RPC.
var ClaudeConfigGet = featureregistry.Feature{
	ID:          "claude-config-get",
	Title:       "Get Claude Config",
	Description: "Retrieves the Claude configuration for a given context.",
	RPCIDs:      []string{"claude-config:get"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

// ClaudeConfigList describes the list-claude-configs RPC.
var ClaudeConfigList = featureregistry.Feature{
	ID:          "claude-config-list",
	Title:       "List Claude Configs",
	Description: "Lists all available Claude configurations.",
	RPCIDs:      []string{"claude-config:list"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

// ClaudeConfigUpdate describes the update-claude-config RPC.
var ClaudeConfigUpdate = featureregistry.Feature{
	ID:          "claude-config-update",
	Title:       "Update Claude Config",
	Description: "Updates the Claude configuration for a given context.",
	RPCIDs:      []string{"claude-config:update"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

func init() {
	featureregistry.Register(ClaudeConfigGet)
	featureregistry.Register(ClaudeConfigList)
	featureregistry.Register(ClaudeConfigUpdate)
}

File: web-app/src/lib/features/features/claude-config.ts
import type { Feature } from '../types';

export const claudeConfigFeatures = {
  'claude-config-get': {
    id: 'claude-config-get',
    title: 'Get Claude Config',
    description: 'Retrieves the Claude configuration for a given context.',
    rpcIds: ['claude-config:get'],
    componentPaths: [],
    testIds: [
      'TestGetClaudeConfig_ReturnsConfig',
      'TestGetClaudeConfig_NotFound',
    ],
    status: 'stable',
    since: '1.0.0',
  },
  'claude-config-list': {
    id: 'claude-config-list',
    title: 'List Claude Configs',
    description: 'Lists all available Claude configurations.',
    rpcIds: ['claude-config:list'],
    componentPaths: [],
    testIds: [
      'TestListClaudeConfigs_ReturnsList',
      'TestListClaudeConfigs_EmptyDir',
    ],
    status: 'stable',
    since: '1.0.0',
  },
  'claude-config-update': {
    id: 'claude-config-update',
    title: 'Update Claude Config',
    description: 'Updates the Claude configuration for a given context.',
    rpcIds: ['claude-config:update'],
    componentPaths: [],
    testIds: [
      'TestUpdateClaudeConfig_NoOp',
    ],
    status: 'stable',
    since: '1.0.0',
  },
} as const satisfies Record<string, Feature>;
